src/session_state.rs

```diff
@@ -31,6 +31,7 @@ pub struct SessionState<
     pub maximum_packet_size: Option<u32>,
     pub client_id: String<64>,
     pub pending_subscriptions: Vec<u16, 32>,
+    pub pending_packets: Vec<u16, MSG_COUNT>,
```

Comment:
This should be combined with `pending_messages` etc. in a single buffer.

Reply:
was about to write that.

Reply:
Actually, MQTT specifies that the client and server have independent sets of packet IDs. Given that this represents the server-provided packet-ids, it is important we separate these. I'll make this more clear with naming etc.

Reply:
Ok. Then `pending_messages()` in `SessionState` and `MqttClient` is ambiguous as it conflates them.
And it may be wise to be very specific and rename the vec to `pending_pubrel`.